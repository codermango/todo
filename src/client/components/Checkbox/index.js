import React from 'react';
import PropTypes from 'prop-types';

const Checkbox = ({ checked, onDone }) => (
  <input
    type="checkbox"
    checked={checked}
    onChange={onDone}
    data-test-id="checkbox"
  />
);

Checkbox.propTypes = {
  checked: PropTypes.bool.isRequired,
  onDone: PropTypes.func.isRequired
};

export default Checkbox;
