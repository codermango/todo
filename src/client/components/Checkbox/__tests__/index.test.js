import React from 'react';
import { shallow } from 'enzyme';
import Checkbox from '../index';


describe('<Checkbox />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Checkbox checked={false} onDone={jest.fn()}/>);
    const instance = wrapper.html();
    expect(instance).toMatchSnapshot();
  });

  it('onChange is called when check box is clicked', () => {
    const wrapper = shallow(<Checkbox checked={false} onDone={jest.fn()} />);
    wrapper.find('input').simulate('change');
    expect(wrapper.props().onChange).toHaveBeenCalledTimes(1);
  });
});
