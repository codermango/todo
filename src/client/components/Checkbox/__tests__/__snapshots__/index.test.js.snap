// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> renders correctly 1`] = `"<input type=\\"checkbox\\" data-test-id=\\"checkbox\\"/>"`;
